src/dmd/declaration.d

```diff
@@ -541,6 +541,11 @@ extern (C++) abstract class Declaration : Dsymbol
         return (storage_class & STC.ref_) != 0;
     }
 
+    final bool isRefOrOut() const pure nothrow @nogc @safe
```

Comment:
I don't think this should be a member of  `declaration`.  Instead consider making a free function like `bool isRefOrOut(StorageClass sc)` so that you can call it like `v.storage_class.isRefOrOut()`.